focus-rune.lic

```diff
@@ -0,0 +1,109 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#focus-rune
+=end
+
+custom_require.call(%w[common-travel common common-items common-arcana spellmonitor])
+
+class Runestone
+  include DRC
+  include DRCT
+  include DRCI
+  include DRCA
+
+  def initialize
+    arg_definitions = [
+      [
+        { name: 'spell',
+          regex: /^[A-z\s\-']+$/i,
+          variable: true,
+          description: "Specify spell to cast from runestone. Spells are: 'Eagle's Cry'
+                'Protection from Evil' Refresh 'Glythtide's Gift' Shadows 'Arc Light'
+                Geyser Bless 'Fire Shards' Althleticism Calm Zephyr Compost
+                'Clear Vision.' Use double quotes." },
+
+        { name: 'debug',
+          regex: /debug/i,
+          optional: true }
+      ]
+    ]
+
+    args = parse_args(arg_definitions)
+    spell = args.spell
+    @debug = args.debug
+    settings = get_settings
+    @runestone_storage = settings.runestone_storage
+    @harness = settings.runestone_harness
+    runestone_purchase = settings.runestone_purchase
+
+    echo spell if @debug
+    echo @harness if debug
+    @rune = spell_name(spell)
+    echo @rune if @debug
+
+    ### Because Protection from Evil has a lowercase letter for the middle word
+    ### this work around is required.
+    if spell == "protection from evil"
+      temp = "Protection from Evil"
+    else
+      temp = spell.split(/ |\_|\-/).map(&:capitalize).join(" ")
+    end
+    echo temp if @debug
+    until DRSpells.active_spells.include?(temp)
+      pause 1 while mana < 40
+      if DRCI.exists?("#{@rune} runestone in #{@runestone_storage}")
+        runestone_cast
+      elsif runestone_purchase
+        buy_runestone
+      else
+        DRC.message("*** Out of #{@rune} runestones! ***") unless runestone_purchase
```

Comment:
I think the `unless` bit can be removed here since per the if/else chain you can't get into this block _unless_ runestone_purchase is already false, correct?
```suggestion
        DRC.message("*** Out of #{@rune} runestones! ***")
```

Reply:
True. When I rewrote this section I did not remove that bit.